web/src/message_list.js

```diff
@@ -109,7 +109,7 @@ export class MessageList {
 
         if (interior_messages.length > 0) {
             this.view.rerender_preserving_scrolltop(true);
-            return true;
```

Comment:
This was weird, and I think incorrect. I tried looking a bit into where this gets used and couldn't find anywhere where `true` made more sense. I'm sort of confused this hasn't caused an issue though.

Reply:
from [here](https://github.com/zulip/zulip/pull/28414/commits/52ce46d7f5a5738c0e2370f7da8dc4ef1aba371b#r1440874862) Tim said:

> Please put this in a separate prep commit -- it seems like a meaningful bug fix?

Reply:
(done)

Reply:
The only use of this return value is here:

https://github.com/zulip/zulip/blob/fe2d7a92b29b9d588b4ea54e58bd61bf891a740e/web/src/message_events.js#L139-L150

This hasn’t been noticed because `true.need_user_to_scroll` isn’t an error in JavaScript, it’s just `undefined`.

Reply:
Seems like it was a small bug though, since it would be falsey instead of truthy